src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -525,22 +525,20 @@ async def setup_datasets(self, pool, uuid):
                         {'properties': update_props_dict},
                     )
 
-            try:
-                await self.middleware.run_in_thread(self.__create_relevant_paths, config.get('create_paths', []))
-            except Exception:
-                self.logger.error('Failed to create relevant paths for %r', dataset, exc_info=True)
-
-    def __create_relevant_paths(self, create_paths):
-        for create_path_config in create_paths:
-            os.makedirs(create_path_config['path'], exist_ok=True)
-            cpath_stat = os.stat(create_path_config['path'])
-            if all(create_path_config[k] for k in ('uid', 'gid')) and (
-                cpath_stat.st_uid != create_path_config['uid'] or cpath_stat.st_gid != create_path_config['gid']
-            ):
-                os.chown(create_path_config['path'], create_path_config['uid'], create_path_config['gid'])
-
-            if (mode := create_path_config.get('mode')) and (cpath_stat.st_mode & 0o777) != mode:
-                os.chmod(create_path_config['path'], mode)
+    def __create_relevant_paths(self, ds_name, create_paths):
+        try:
+            for create_path_config in create_paths:
```

Comment:
The `try: except:` should be inside the `for` loop so we don't crash on the first path and ignore the others.